scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -526,10 +550,42 @@
      :backfill-jobs (mapv :job/uuid jobs-to-backfill)
      :matched-head? (matched? head-of-considerable)}))
 
+(defn task->mesos-message
+  [t]
+  (let [cmd (:command t)
+        resources (:resources t)]
+    {:name (:name t)
+     :task-id [{:value (:task-id t)}]
+     :slave-id [{:value (:slave-id t)}]
+     :resources (vec (remove nil?
+                             [{:type :value-scalar,
+                               :name "mem",
+                               :scalar (:mem resources)
+                               :role (:role t)}
+                              {:type :value-scalar,
+                               :name "cpus",
+                               :scalar (:cpus resources)
+                               :role (:role t)}
+                              (when (seq (:ports resources))
+                                {:type :value-ranges,
+                                 :name "ports",
+                                 :ranges (:ports resources)
+                                 :role (:role t)})]))
+
+     :labels {:labels (mapv (fn [kv] {:key (key kv) :value (val kv)})
+                            (vec (:labels t)))}
+     :command {:value (:value cmd)
+               :environment {:variables
```

Comment:
Perhaps use an `update-in` for environment instead of destructing and restructuring the whole map